packages/nvidia_nat_core/src/nat/front_ends/fastapi/fastapi_front_end_plugin_worker.py

```
@@ -1263,147 +1263,165 @@ class MCPToolInfo(BaseModel):
+        async def get_mcp_client_tool_list() -> MCPClientToolListResponse:
+            """
+            Get the list of MCP tools from all MCP clients in the workflow configuration.
+            Checks session health and compares with workflow function group configuration.
+            """
+            try:
+                # Get all function groups from the builder
+                function_groups = {name: cfg.instance for name, cfg in builder._function_groups.items()}
+                mcp_clients_info = await _collect_mcp_client_tool_list(function_groups)
                 return MCPClientToolListResponse(mcp_clients=mcp_clients_info)
 
             except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)